lldb/docs/resources/lldbgdbremote.md

```diff
@@ -735,6 +735,56 @@ This is a performance optimization, which speeds up debugging by avoiding
 multiple round-trips for retrieving thread information. The information from this
 packet can be retrieved using a combination of `qThreadStopInfo` and `m` packets.
 
+### MultiMemRead
+
+Read memory from multiple memory ranges.
+
+This packet has one argument:
+
+* `ranges`: a list of pairs of numbers, formatted in base-16. Each pair is
+separated by a `,`, as is each number in each pair. The first number of the
+pair denotes the base address of the memory read, the second denotes the number
+of bytes to be read. The list must end with a `;`.
+
+The reply packet starts with a comma-separated list of numbers formatted in
+base-16, denoting how many bytes were read from each range, in the same order
+as the request packet. The list is followed by a `;`, followed by a sequence of
+bytes containing binary encoded data for all memory that was read. The length
+of this sequence must be equal to the sum of the numbers provided at the start
+of the reply. The order of the binary data is the same as the order of the
+ranges in the request packet.
+
+If an entire range is not readable, the stub may perform a partial read of a
+prefix of the range.
+
+If the stub is unable to read any bytes from a particular range, it must return
```

Comment:
This is a bit ambiguous because "unable to read any" could mean they all failed or just one.

You need a way to say "if by applying the rules above, the stub has read 0 bytes from the range...".